app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java

```diff
@@ -114,14 +114,16 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     protected void createGoogleApiClient() {
-        mGoogleApiClient = new GoogleApiClient.Builder(this)
-                .addApi(Plus.API)
-                .addApi(Games.API)
-                .addApi(AppStateManager.API)
-                .addScope(Plus.SCOPE_PLUS_LOGIN)
-                .addScope(Plus.SCOPE_PLUS_PROFILE)
-                .addScope(Games.SCOPE_GAMES)
-                .build();
+        GoogleApiClient.Builder builder = new GoogleApiClient.Builder(this)
+                .addApiIfAvailable(AppStateManager.API);
+        if (PrefUtils.isSignedIn(this)) {
+            builder.addApiIfAvailable(Plus.API, Plus.SCOPE_PLUS_LOGIN, Plus.SCOPE_PLUS_PROFILE)
```

Comment:
The client is built when creating the GDG activity.
On 13 Dec 2015 10:17, "Said Tahsin Dane" <notifications@github.com> wrote:

> In app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java
> <https://github.com/gdg-x/frisbee/pull/501#discussion_r47442171>:
>
> > @@ -114,14 +114,16 @@ public void onCreate(Bundle savedInstanceState) {
> >      }
> >
> >      protected void createGoogleApiClient() {
> > -        mGoogleApiClient = new GoogleApiClient.Builder(this)
> > -                .addApi(Plus.API)
> > -                .addApi(Games.API)
> > -                .addApi(AppStateManager.API)
> > -                .addScope(Plus.SCOPE_PLUS_LOGIN)
> > -                .addScope(Plus.SCOPE_PLUS_PROFILE)
> > -                .addScope(Games.SCOPE_GAMES)
> > -                .build();
> > +        GoogleApiClient.Builder builder = new GoogleApiClient.Builder(this)
> > +                .addApiIfAvailable(AppStateManager.API);
> > +        if (PrefUtils.isSignedIn(this)) {
> > +            builder.addApiIfAvailable(Plus.API, Plus.SCOPE_PLUS_LOGIN, Plus.SCOPE_PLUS_PROFILE)
>
> For News feed actually, we cannot unfortunately use GoogleApiClient. It's
> because it is really limited. We have another dependency for Google plus.
>
> I am not sure but this comes to my mind. You are not adding those scopes
> for non logged in users. What happens when they click login? How do you add
> these scopes later?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gdg-x/frisbee/pull/501/files#r47442171>.
>